modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -459,14 +474,29 @@ TensorMechanicsAction::actOutputGeneration()
   // Add variables (optional)
   if (_current_task == "add_aux_variable")
   {
-    auto params = _factory.getValidParams("MooseVariableConstMonomial");
-    params.set<MooseEnum>("order") = "CONSTANT";
-    params.set<MooseEnum>("family") = "MONOMIAL";
-    // Loop through output aux variables
+    unsigned int index = 0;
+
     for (auto out : _generate_output)
     {
-      // Create output helper aux variables
-      _problem->addAuxVariable("MooseVariableConstMonomial", _base_name + out, params);
+      if (_material_output_family[index] == "MONOMIAL")
+      {
+        // Create output helper aux variables
+        auto params = _factory.getValidParams("MooseVariableConstMonomial");
```

Comment:
`MooseVariableConstMonomial` might not be the right choice for non CONSTANT monomials

Reply:
Yeah, originally I had the if statement tying CONSTANT with MONOMIAL.  I will add additional conditions